### PR TITLE
GraphQL: Improve schema error handling & tidy

### PR DESCRIPTION
**Pull request by MetRonnie**
Originally created this branch to deal with #6023, however ended up fixing that in the uiserver at cylc/cylc-uiserver#566

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes 
- [x] No tests needed
- [x] No changelog needed
- [x] No docs needed
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/network/schema.py [hunk 1]
```diff
@@ -116,22 +126,6 @@ def sort_elements(elements, args):
...
-    'task_proxy',
-]
-
-PROXY_TYPES = [
-    'family_proxies',
-    'family_proxy',
-    'task_proxies',
-    'task_proxy',
-]
-
```
Comment (MetRonnie) [inline cylc/flow/network/schema.py:134]: These were apparently unused

File: cylc/flow/network/schema.py [hunk 2]
```diff
@@ -283,22 +277,35 @@ class SortArgs(InputObjectType):
 
 # Resolvers:
 
-def get_type_str(obj_type):
-    """Iterate through the objects of_type to find the inner-most type."""
-    pointer = obj_type
-    while hasattr(pointer, 'of_type'):
-        pointer = pointer.of_type
-    return str(pointer).replace('!', '')
```
Comment (MetRonnie) [inline cylc/flow/network/schema.py:291]: `graphql-core` comes with a function that does this https://github.com/graphql-python/graphql-core-legacy/blob/v2.3.2/graphql/type/definition.py#L54-L60

**Comment by oliver-sanders** [comment]
Bumping milestone for prioritisation, change back when merged.

**Comment by MetRonnie** [comment]
@dwsutherland Could you review this? No worries if you're too busy

File: cylc/flow/network/schema.py (hunk 1 above)
Comment (dwsutherland) [reply cylc/flow/network/schema.py:134]: Must have developed past them, where used in resolvers

**Review by dwsutherland — APPROVED** [review]
LGTM, had a play and tests working 👍

**Pull request merged**